vm/src/exceptions.rs

```diff
@@ -859,6 +860,24 @@ fn key_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
     }
 }
 
+fn os_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
+    let args = exc.args();
+    if args.as_slice().len() == 2 {
+        // SAFETY: len() == 2 is checked so get_arg 1 or 2 won't panic
+        let s = match (
+            exc.get_arg(0).unwrap().str(vm),
+            exc.get_arg(1).unwrap().str(vm),
+        ) {
+            (Ok(errno), Ok(msg)) => format!("[Errno {}] {}", errno, msg),
+            (Ok(errno), Err(_)) => format!("[Errno {}]", errno),
+            (Err(_), _) => "Error".to_owned(),
+        };
```

Comment:
Ah, this now handles the error if `__str__` fails, which I don't believe should happen, i.e:

```python
class Foo:
    def __str__(self):
        raise TypeError("for whatever reason")

OSError(Foo(), Foo()).__str__()  # doesn't raise
```

The match isn't needed, just use `?` on `str` and use the result directly, something like:

```rust
let s = format!("[Error {}] {}", exc.get_arg(0).unwrap().str(vm)?, exc.get_arg(1).unwrap().str(vm)?;
```

Reply:
Interesting. Maybe `str` can return `PyStrRef` instead of `PyResult<PyStrRef>`.
I updated the code as suggested.